specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+    // HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND).
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions(
+        nullptr, m_userDataFolder.c_str(), options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            this, &AppWindow::OnCreateEnvironmentCompleted)
+            .Get());
+}
+void AppWindow::OnCreateEnvironmentCompleted(
+    HRESULT result, ICoreWebView2Environment* environment)
+{
+    if (environment)
+    {
+        wil::unique_cotaskmem_string version;
+        CHECK_FAILURE(environment->get_BrowserVersionString(&version));
+        CheckChannel(version.get());
+    }
+}
+```
+# API Details
+```
+/// The WebView2 Runtime channel. Use `RuntimeChannelPreferenceSet` and
+/// `ShouldReverseChannelSearchOrder` on `ICoreWebView2EnvironmentOptions` to
+/// control which channel the WebView2 loader searches for.
+/// |Channel|Primary purpose|How often updated with new features|
+/// |:---:|---|:---:|
+/// |Stable (WebView2 Runtime)|Broad Deployment|~4 weeks|
+/// |Beta|Flighting with inner rings, automated testing|~4 weeks|
+/// |Dev|Automated testing, selfhosting to test new APIs and features|Weekly|
+/// |Canary|Automated testing, selfhosting to test new APIs and features|Daily|
+typedef enum COREWEBVIEW2_RUNTIME_CHANNEL {
+  /// The stable WebView2 Runtime that is released every 4 weeks.
+  COREWEBVIEW2_RUNTIME_CHANNEL_STABLE,
+  /// The Beta Runtime channel that is released every 4 weeks, a week before the
+  /// stable release.
+  COREWEBVIEW2_RUNTIME_CHANNEL_BETA,
+  /// The Dev Runtime channel that is released weekly.
+  COREWEBVIEW2_RUNTIME_CHANNEL_DEV,
+  /// The Canary Runtime channel that is released daily.
+  COREWEBVIEW2_RUNTIME_CHANNEL_CANARY,
+} COREWEBVIEW2_RUNTIME_CHANNEL;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_RUNTIME_CHANNEL)")
+
+/// Additional options used to create the WebView2 Environment that support
+/// specifying the `RuntimeChannelPreferenceSet` and `ReverseChannelSearchOrder`.
+[uuid(47ac856e-a726-4e04-b36b-f58dafb39e38), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+
+  /// Gets the `RuntimeChannelPreferenceSet`.
+  HRESULT GetRuntimeChannelPreferenceSet(
+      [out] COREWEBVIEW2_RUNTIME_CHANNEL* channels);
+
+  /// Sets the `RuntimeChannelPreferenceSet`, which is a mask of one or more
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s. OR operation(s) can be applied to multiple
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s to create a mask. The default value is a
+  /// a mask of all the channels. By default, the WebView2 loader
+  /// searches for channels from most to least stable, using the first channel
+  /// found on the device. When a `RuntimeChannelPreferenceSet` is provided, the
+  /// WebView2 loader will only search for the channels specified in the set.
+  /// Set `ReverseChannelSearchOrder` to `TRUE` to reverse the search order so
```

Comment:
It feels like `ReverseChannelSearchOrder` could also be implemented by trying the channels in the order that you want until you get the answer you are looking for.  That property doesn't fully make sense to me as to when I would want it.  I feel like I would either not care (default options) or I would want to specify an exact set.  Reversing the order means the results are less deterministic.

Reply:
If I was doing automated testing I would probably want to force specific runtimes.  I'd have a "Beta" test pass that forces the "Beta" WV2 runtime.  I'd have equivalent Canary/Dev tests as well.  I don't think I would want "try for Canary but take what you can get, which might be Stable".